controllers/restore_controller.go

```diff
@@ -19,133 +19,370 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"sort"
 	"time"
 
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
 	"github.com/pkg/errors"
-	veleroapi "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
-	k8serr "k8s.io/apimachinery/pkg/api/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
 	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	v1beta1 "github.com/open-cluster-management/cluster-backup-operator/api/v1beta1"
+	veleroapi "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
 )
 
 var (
-	restoreRequeueInterval = time.Minute * 1
+	restoreOwnerKey = ".metadata.controller"
+	apiGVStr        = v1beta1.GroupVersion.String()
 )
 
+const (
+	managedClusterImportInterval            = 10 * time.Second // as soon restore is finished we start to poll for managedcluster registration
+	BootstrapHubKubeconfigSecretName        = "bootstrap-hub-kubeconfig"
+	OpenClusterManagementAgentNamespaceName = "open-cluster-management-agent" // TODO: double check whether this can change
```

Comment:
The ACM controller can be installed in a namespace different then the `open-cluster-management` which is the default; in this case the restore is not going to work

The backup operator finds the ACM install namespace by looking for the charts-v1 channel, which is created in the acm ns
I am not sure if this is the right approach but charts-v1 seems to be a constant for now

Reply:
Is this valid for the managed cluster as well?

Reply:
No, this is working for the hubs only .. where acm is installed

Reply:
OK, thanks but this one is the `managedclsuter ACM` namespace. Need to find a way to retrieve this info. Thanks for rising up this

Reply:
got it .. maybe we can have an optional used property to define the ACM namespace ( as we did with the velero namespace on the VeleroConfigBackupProxy ); if not defined is open-cluster-management

Reply:
@birsanv I found _the right solution_ for this. The information is stored in the cluster scoped `klusterlet.spec.namespace`.  So this is going to be ~~`OpenClusterManagementAgentNamespaceName`~~ will become `DefaultOpenClusterManagementAgentNamespaceName`. Most of the time the default is kept, so in order to avoid an useless remote REST call I'll try the default and eventually I'll get the non default info through a REST call. Many thanks to @qiujian16  for the info.